ibacm/src/acm.c

```diff
@@ -2596,17 +2596,64 @@ acm_open_port(struct acmc_port *port, struct acmc_device *dev, uint8_t port_num)
 	port->state = IBV_PORT_DOWN;
 }
 
+static bool dev_support_ib(struct ibv_device *ibdev)
+{
+	struct ibv_context *verbs;
+	struct ibv_device_attr attr;
+	struct ibv_port_attr port_attr;
+	int i, ret;
+	bool has_ib_port = false;
+
+	verbs = ibv_open_device(ibdev);
```

Comment:
Let us try to avoid doing open_device twice?

Reply:
I think that do open_device twice can keep the code simple. BTW, ibv_query_port also had been do twice. The function is not in critical path, so I think do twice is OK.

If you have any suggestion, please feel free to edit the patch.

Thanks